docs/DeveloperGuide.md

```diff
@@ -1,38 +1,332 @@
 # Developer Guide
-
+<!----------------------------Table of Contents----------------------------------------->
+- [Acknowledgements](#acknowledgements)
+- [Design](#design)
+  - [Architecture](#architecture)
+  - [UI Component](#ui-component)
+  - [Logic Component](#logic-component)
+  - [Model Component](#model-component)
+  - [Storage Component](#storage-component)
+  - [Common classes](#common-classes)
+- [Implementation](#implementation)
+  - [Add Module](#add-module)
+  - [Remove Module](#remove-module)
+  - [Mark Module As Taken](#mark-module-as-taken)
+  - [List Modules](#list-modules)
+  - [Display Status](#display-status)
+  - [Get Module Details](#get-module-details)
+  - [Initialise User](#initialise-user)
+  - [Save To Local Drive](#save-plan-to-local-drive)
+- [Documentation, logging, testing, configuration, dev-ops](#documentation-logging-testing-configuration-dev-ops)
+  - [Documentation](#documentation)
+  - [Logging](#logging)
+  - [Testing](#testing)
+  - [Configuration](#configuration)
+  - [Dev-ops](#dev-ops)
+- [Appendix A: Requirements](#appendix-a--requirements)
+- [Appendix B: Instructions for manual testing](#appendix-b--instructions-for-manual-testing)
+<!----------------------------Acknowledgements----------------------------------------->
 ## Acknowledgements
-
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+<!----------------------------Design----------------------------------------->
+## Design
+
+### Architecture
+
+![ArchitectureDiagram](uml/diagrams/Architecture.png)
+
+### UI Component
+![UIClassDiagram](uml/diagrams/UiClass.png)
```

Comment:
Neat UI component! I think Scanner and PrintStream may not be needed (if im not wrong)